### ISSUE TITLE
mmdc fails to parse .dsl mermaid file generated with tsuml2

### ISSUE THREAD

**Issue by hamoid** [demike/TsUML2#95]
Hi! I'm generating a mermaid file with

```
node node_modules/tsuml2/dist/bin/index.js --memberAssociations --glob "./src/**/*.ts" --outMermaidDsl "out.dsl"
```

It finishes correctly, but when I try to convert it to an image with `mmdc -i out.dsl -o out2.svg` I get the following

```
Generating single mermaid chart

Error: Parse error on line 28:
...Type) =~ void-cfg: { seed: number; desi
----------------------^
Expecting 'STRUCT_STOP', 'MEMBER', got 'OPEN_IN_STRUCT'
```

This is the content of the generated file:

```
classDiagram

class CoreShape{
            +container: Object3D~Object3DEventMap~
+baseModel: Object3D~Object3DEventMap~
+settings: Settings
+textures: Textures
+antennas: Antenna[]
+cables: HangingCable[]
+ground?: Ground
            +generate() void
        }
CoreShape  --  Settings
CoreShape  -- "0..*" Antenna
CoreShape  -- "0..*" HangingCable
CoreShape  --  Ground
class Exporter{
            -exporter: GLTFExporter
-downloadLink: HTMLAnchorElement
            -save() void
-saveString() void
-saveArrayBuffer() void
+exportObject() void
+exportScene() void
        }
class Settings{
            +onChange: (ct: ChangeType) =~ void
-cfg: { seed: number; designType: string; baseModel: string; }
-folder: GUI
+cableResolution: number
+maxCableCount: number
+maxAntennaFailedAttempts: number
+maxAntennaCount: number
            +incSeed() void
        }
class ChangeType {
        <<enumeration>>
        seed
designType
baseModel
      }
class SocketChooser{
            -ground: Ground
-settings: Settings
+sockets: Socket[]
-chosenTypes: any
-randomId: number
-pointToface?: Vector3
            +add() void
+getUnusedSocket() Socket
+getAntennaType() number
        }
SocketChooser  --  Ground
SocketChooser  --  Settings
SocketChooser  -- "0..*" Socket
class Antenna{
            +instanceID: symbol
+typeID: number
#baseSocket: Socket
#childSockets: Socket[]
+OBBs: OrientedBoundingBox[]
+group: Group~Object3DEventMap~
            +getBaseSocket() Socket
+getChildSockets() Socket[]
+dispose() void
+computeDebug() void
-disposeGroup() void
        }
class IAntenna {
            <<interface>>
            +instanceID: symbol
+typeID: number
            +getChildSockets() Socket[]
        }
IAntenna<|..Antenna
Antenna  --  Socket
Antenna  -- "0..*" Socket
Antenna  -- "0..*" OrientedBoundingBox
class CellPhoneAntenna{
            
            
        }
Antenna<|--CellPhoneAntenna
class HangingCable{
            +group: Group~Object3DEventMap~
+cablePath?: CatenaryCurve
+OBB?: OrientedBoundingBox
            -createMeshFromCatenaryCurve() void
+dispose() void
-disposeGroup() void
        }
class HangingCableParameters {
            <<interface>>
            +minRadius?: number
+maxRadius?: number
+pointCount?: number
+scaleFactor?: number
            
        }
HangingCable  --  CatenaryCurve
HangingCable  --  OrientedBoundingBox
class Spiral{
            -axis: Socket
-startRadius: number
-endRadius: number
-turns: number
-angleOffset: number
            +getRadius() number
+getPoint() Vector3
        }
class Helix{
            
            
        }
Curve~TVector~<|--Spiral
Antenna<|--Helix
Spiral  --  Socket
class HornAntenna{
            
            
        }
Antenna<|--HornAntenna
class InsectAntenna{
            
            
        }
Antenna<|--InsectAntenna
class MastAntenna{
            
            
        }
Antenna<|--MastAntenna
class Microwave{
            
            +createCylinder() void
+createClips() void
        }
class MicrowaveParameters {
            <<interface>>
            +cylinderRadius?: number
+antennaHeight?: number
+coneHeight?: number
+radialSegments?: number
+clipCount?: number
            
        }
Antenna<|--Microwave
class ParabolicAntenna{
            
            +generateFeedAntenna() void
        }
class ParabolicAntennaParameters {
            <<interface>>
            +radius?: number
+steps?: number
+thickness?: number
+aperture?: number
+isWireframe?: boolean
            
        }
Antenna<|--ParabolicAntenna
class Radome{
            
            +createDome() void
+createBase() void
        }
class RadomeParameters {
            <<interface>>
            +sphereRadius?: number
+sphereResolution?: number
+wireframeThickness?: number
+baseSegments?: number
+baseHeight?: number
            
        }
Antenna<|--Radome
class TowerAntenna{
            
            
        }
Antenna<|--TowerAntenna
class YagiUdaAntenna{
            -material: Material
            +createDirectionalAntenna() void
+createDirectorArray() void
+createReflector() void
+createRefrectorArray() void
        }
class YagiUdaAntennaParameters {
            <<interface>>
            +arrayDistance?: number
+coreDiameter?: number
            
        }
Antenna<|--YagiUdaAntenna
class TSAntennaType {
            <<type>>
            
            
        }
class StorableOrbitControls{
            
            +loadLocalState() void
+saveLocalState() void
        }
OrbitControls<|--StorableOrbitControls
class CatenaryCurve{
            -len: number
-dy: number
-mx: number
-a: number
-from: Vector3
-to: Vector3
            +getPoint() Vector3
        }
Curve~TVector~<|--CatenaryCurve
class Ground{
            -sockets: Socket[]
+mesh: Mesh~BufferGeometry~NormalBufferAttributes~, Material | Material[], Object3DEventMap~
+OBBs: OrientedBoundingBox[]
+noiseScale: number
            +getSockets() Socket[]
-buildLandscapeMesh() void
-addVerticalSocket() void
        }
Ground  -- "0..*" Socket
Ground  -- "0..*" OrientedBoundingBox
class Socket{
            +start: Vector3
+end: Vector3
+owner: Antenna
+depth: number
+allowsCable: boolean
+normal: Vector3
+_used: boolean
+length: number
+matrix: Matrix4
+socketHelper?: SocketHelper
+collectionType: SocketCollectionType
            +update() void
+alignedChild() Socket
+childLength() number
+addToDebugScene() void
+setCollectionType() void
+getCollectionType() SocketCollectionType
        }
Socket  --  SocketHelper
class OrientedBoundingBox{
            +debugCube: Mesh~BufferGeometry~NormalBufferAttributes~, Material | Material[], Object3DEventMap~
#connectedToID?: symbol
            +show() void
+hide() void
+addToDebugScene() void
+connectBySymbol() OrientedBoundingBox
+isConnected() boolean
+dispose() void
        }
OBB<|--OrientedBoundingBox
class SocketHelper{
            
            +show() void
+hide() void
+setUsed() void
+addToDebugScene() void
        }
ArrowHelper<|--SocketHelper
```

If I generate an SVG file instead it works. I'm trying with mermaid because the default SVG output has too many lines crossing objects, which I find hard to parse visually.

**Comment by demike** [comment]
@hamoid thanks for reporting! Should be fixed now by escaping the "{" and "}"
